cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/RemoteDatasetFramework.java

```diff
@@ -199,6 +203,7 @@ private void addModule(Id.DatasetModule moduleId, Class<?> typeClass) throws Dat
     try {
       File tempFile = File.createTempFile(typeClass.getName(), ".jar");
       try {
+        // TODO: should be namespaced?
```

Comment:
you are probably right. This is different from what I initially thought. Removing the TODO.